### PR TITLE
fix(images): enforce image props + fix some bugs

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-864] 

### Description

A while back we updated our underlying image component and this led to a few bugs because the newer component was less tolerant of missing props. 
https://artsy.slack.com/archives/C02BAQ5K7/p1714588769205339 conversation with amber on the latest got me thinking we might be able to enforce this in the types and catch these.

This pr:
- adds some type enforcement to the OpaqueImageView component
- updates components that were passing width and height in styles to pass in dedicated height and width props (these were rendering fine but limits to what we can check on style props so just getting rid of)
- fixes article autosuggest results image not rendering caused by useRawURL logic bug when using palette image
- fixes show card not rendering when only one image for a show
- fixes showPreview card not rendering in conversations (see this convo if you are surprised that this exists like me: https://artsy.slack.com/archives/C05F8TNKGAV/p1714675431675669
- fixes image render in view in room artwork details
- fixes margin not being applied in artist shows
- deletes unused code related to Trove and ShowsList

#### Articles Autosuggest

 <details>
  <summary>
   Articles autosuggest before
  </summary>

articles-suggest-before

 </details>

 <details>
  <summary>
   Articles autosuggest after
  </summary>

articles-suggest-after

 </details>

#### Show Card

 <details>
  <summary>
   Show Card before
  </summary>

show-card-before

 </details>

 <details>
  <summary>
    Show Card after
  </summary>

show-card-after

 </details>

#### Artist Shows

 <details>
  <summary>
   Artist Shows before
  </summary>

shows-before

 </details>

 <details>
  <summary>
  Artist Shows after
  </summary>

shows-after

 </details>

#### Viewing Room Image Details

 <details>
  <summary>
   Viewing Room Image Detailst before
  </summary>

viewing-room-image-before

 </details>

 <details>
  <summary>
   Viewing Room Image Details after
  </summary>

viewing-room-image-after

 </details>

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Fix image rendering issues - brian 

#### iOS user-facing changes

- 

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-864]: https://artsyproduct.atlassian.net/browse/PHIRE-864?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx [hunk 1]
```diff
@@ -7,7 +7,36 @@ import React, { useCallback, useRef, useState } from "react"
...
+  width: number
+  height?: never
+}
+
+type WithHeightAndWidth = {
+  height: number
+  width: number
+}
+
+type DimensionProps = AspectRatioProps & (WithHeight | WithWidth | WithHeightAndWidth)
```
Comment (brainbicycle) [inline src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx:37]: type magic courtesy of chatGPT, support (aspectRatio with height, aspectRatio with width, width and height) don't support (aspectRatio alone, width without height or aspectRatio, etc.)

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Fix image rendering issues - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b10a31a5387b1fee2f7ad1b7bb9fcc1b90263950
</p>

**Review by jpotts244 — APPROVED** [review]
Very nice!! 🏅

File: src/app/Components/Artist/ArtistShows/VariableSizeShowsList.tsx
Comment (brainbicycle) [inline src/app/Components/Artist/ArtistShows/VariableSizeShowsList.tsx:1]: dead code as far as I can tell

File: src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx [hunk 2]
```diff
@@ -178,7 +197,7 @@ export const OpaqueImageView: React.FC<Props> = ({ aspectRatio, ...props }) => {
       <Image
         src={props.imageURL}
         geminiResizeMode={aspectRatio ? "fit" : "fill"}
-        performResize={props.useRawURL}
+        performResize={!props.useRawURL}
```
Comment (brainbicycle) [inline src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx:200]: little bug 🐛

File: src/app/Components/Home/HeroUnit.tsx
Comment (brainbicycle) [inline src/app/Components/Home/HeroUnit.tsx:1]: this was only used for Trove as far as I can tell and replace by a new HeroUnit component for other heroUnits

**Review by araujobarret — APPROVED** [review]

**Review by MounirDhahri — APPROVED** [review]

**Pull request merged**